packages/e2e-tests/helpers/new-react-devtools-panel.ts

```diff
@@ -32,6 +105,98 @@ export async function openReactDevtoolsPanel(page: Page) {
   await openReactDevtoolsPanelOld(page);
 }
 
+export async function searchComponents(page: Page, text: string) {
+  const input = getSearchInput(page);
+
+  await waitFor(async () => {
+    await expect(await input.isEnabled()).toBe(true);
+  });
+
+  await waitFor(async () => {
+    await input.focus();
+    await input.press("Escape"); // Clear/reset
+    await input.type(text);
+    await input.press("Enter");
+
+    const searchResults = page.locator('[data-test-id="ReactDevTools-SearchResults"]');
+    await searchResults.waitFor();
+  });
+}
+
+export async function verifyInspectedPropertyValue(
+  page: Page,
+  category: "Hooks" | "Props",
+  name: string,
+  expectedValue: string
+): Promise<void> {
+  await debugPrint(
+    page,
+    `Expect property "${name}" to have value "${expectedValue}"`,
+    "verifyInspectedPropertyValue"
+  );
+
+  const section =
+    category === "Hooks"
+      ? page.locator('[data-test-id="ReactDevTools-Section-Hooks"]')
+      : page.locator('[data-test-id="ReactDevTools-Section-Props"]');
+
+  const keyValue = section.locator(`[data-test-name="KeyValue"]`, { hasText: name });
+  const value = keyValue.locator('[data-test-name="ClientValue"]');
+  const actualValue = await value.textContent();
+
+  expect(actualValue).toBe(expectedValue);
+}
+
 export async function verifyReactComponentIsSelected(component: Locator) {
   await expect(component).toHaveAttribute("data-selected", "true");
 }
+
+export async function verifySearchResults(
+  page: Page,
+  expectations: {
+    currentNumber?: number;
+    name?: string;
+    totalNumber?: number;
+  }
+) {
+  const {
+    currentNumber: expectedCurrent,
+    name: expectedName,
+    totalNumber: expectedTotal,
+  } = expectations;
+
+  if (expectedCurrent != null || expectedTotal != null) {
+    await waitFor(async () => {
+      const { current: actualCurrent, total: actualTotal } =
+        (await getComponentSearchResultsCount(page)) ?? {};
+
+      if (expectedCurrent != null) {
+        expect(actualCurrent).toBe(expectedCurrent);
+      }
+
+      if (expectedTotal != null) {
+        expect(actualTotal).toBe(expectedTotal);
+      }
+    });
+  }
+
+  if (expectedName != null) {
+    await waitFor(async () => {
+      await verifySelectedComponentName(page, expectedName);
```

Comment:
I don't think this `waitFor()` makes sense because `verifySelectedComponentName()` has a `waitFor()` itself.

Reply:
That's fair. I think I probably wrote this wait-for before moving that stuff into a helper function.